test/test_nestedtensor.py

```diff
@@ -1069,6 +1220,40 @@ def test_size_dim(self):
             RuntimeError, "Given dimension 1 is irregular and does not have a size", lambda: a.size(1))
         self.assertEqual(a.size(2), 4)
 
+    def test_nested_tensor_matmul(self):
+        a = torch.randn(2, 6, requires_grad=True, dtype=torch.float64)
+        b = torch.randn(3, 6, requires_grad=True, dtype=torch.float64)
+        c = torch.randn(6, 4, requires_grad=True, dtype=torch.float64)
+        d = torch.randn(6, 5, requires_grad=True, dtype=torch.float64)
+
+        def grad_test_func(a, b, c, d):
+            nt0 = torch.nested_tensor([a, b])
+            nt1 = torch.nested_tensor([c, d])
+            result = torch.matmul(nt0, nt1)
+            return result.to_padded_tensor(0.0)
+
+        data = (a, b, c, d)
+        assert torch.autograd.gradcheck(grad_test_func, inputs=data)
+
+    def test_nested_tensor_matmul_backward(self):
+        nt0 = torch.nested_tensor([torch.randn((2, 6)), torch.randn((3, 6))])
+        nt1 = torch.nested_tensor([torch.randn((6, 4)), torch.randn((6, 5))])
+        pt0 = nt0.to_padded_tensor(0.0)
+        pt1 = nt1.to_padded_tensor(0.0)
+
+        nt0.requires_grad_(True)
+        nt1.requires_grad_(True)
+        pt0.requires_grad_(True)
+        pt1.requires_grad_(True)
```

Comment:
The reason I did this is to avoid gradient accumulation.

Concretely, if I set requires_grad in the nt factory line, then `to_padded_tensor` will be recorded in the computation graph, so the backward from pt will go all the way down to nt and ruin nt.grad

Reply:
can you get around this using `with torch.no_grad()` for `the to_padded_tensor()` conversions?

Reply:
yes it works